File: Lista3/Lista3_11800910/exec3/Quadrado.java
/**
 * Classe que cria o quadrado
 */
public class Quadrado extends Retangulo {

    /**
     * Retângulo com ambos os lados iguais
     * @param lado
     * @param cheio
     * @param cor
     */
    public Quadrado(double lado, boolean cheio, String cor){
        super(lado, lado, cheio, cor);
    }

    /**
     * toString
     */
    @Override
    public String toString(){
        boolean cheio = getCheio();
        String s = "\nForma Geométrica: QUADRADO\n";
        s += "Lados: " + getLados()[0] + "\n";
        s += "Area: " + area() + "\n";
        s += "Perímetro: " + perimetro() + "\n";
        s += "Colorido: " + cheio + "\n";
        if(cheio){ 
            s += "Cor: " + getCor() + "\n"; 
        }
        return s;
    }

}

File: Lista4/exec3/quadrado.py
from forma import *

class quadrado(forma):

    def __init__(self, lado: float, cheio: bool, cor: str) -> None:
        super().__init__(lado, lado, cheio, cor)

    def __str__(self) -> str:
        self.cheio = self.getCheio()
        s = '\nForma Geométrica: QUADRADO\n'
        s += "Lados: " + str(self.getLados()[0]) + " e " + str(self.getLados()[1]) + "\n"
        s += "Area: " + str(self.area()) + "\n"
        s += "Perímetro: " + str(self.perimetro()) + "\n"
        s += "Colorido: " + str(self.cheio) + "\n"
        if(self.cheio):
            s += "Cor: " + self.getCor() + "\n"
        
        return s

